ethereum/rpc/namespaces/eth/api.go

```diff
@@ -567,18 +577,28 @@ func (e *PublicAPI) doCall(
 }
 
 // EstimateGas returns an estimate of gas usage for the given smart contract call.
-func (e *PublicAPI) EstimateGas(args evmtypes.CallArgs) (hexutil.Uint64, error) {
+func (e *PublicAPI) EstimateGas(args evmtypes.CallArgs, blockNrOptional *rpctypes.BlockNumber) (hexutil.Uint64, error) {
```

Comment:
This is supposed to be `blockNrOrHash *rpc.BlockNumberOrHash` right?

Reply:
Yeah, but in our code base, there are several other apis that use `rpctypes.BlockNumber` which is actually `BlockNumberOrHash` in go-ethereum side. Maybe we can do that together in another PR.